File: ruby/classEachW/prueba.rb
#!/usr/bin/env ruby

class Animal
  attr_accessor :name
  def initialize(name)
    @name = name
    puts "Soy un #{name}"
  end
end

animals = %w(gato perro tiburon leon pecesito hamster)
objetos = []
conta = 0

animals.each do |animal|
  objetos[conta] = Animal.new(animal)
  objetos[conta].name = 'Ahora soy un animal llamado '+objetos[conta].name
  conta +=1
end

objetos.each do |o|
  puts o.name
end

File: java/HerenciaAnimal/Prueba.java
class Animal{
	public Animal(){
		System.out.println("Soy un animal");
	}
}

class Mamifero extends Animal{
	public Mamifero(){
		System.out.println("Soy un mamifero");
	}
}

class Perro extends Mamifero{
	public Perro(){
		System.out.println("Soy un perro");
	}
}

public class Prueba{
	public static void main(String[] args){
		Animal dog = new Perro();
	}
}
